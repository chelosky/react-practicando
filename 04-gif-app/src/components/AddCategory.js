import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories  }) => {

    const [value, setValue] = useState('');

    const handleInputChange = (e) => {
        setValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(value.trim().length > 2){
            setCategories( item => [ value, ...item ] );
            setValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ value }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}