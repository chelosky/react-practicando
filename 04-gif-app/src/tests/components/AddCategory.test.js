import React from "react"

import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    } )

    test('should created component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should input changed', () => {
        const input = wrapper.find('input');
        const value = 'ejemplo';
        input.simulate('change', { target: { value } });
        expect( wrapper.find('input').prop('value')).toBe(value);
    })
    
    test('should form submit when value void', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('should form submit when valud input is not void', () => {
        const input = wrapper.find('input');
        const value = 'ejemplo';
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect( wrapper.find('input').prop('value')).toBe('');
    })
    
})