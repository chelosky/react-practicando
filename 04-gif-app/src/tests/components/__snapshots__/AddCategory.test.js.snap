// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should created component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
